File: components/credentials/src/main/java/org/cloudfoundry/credhub/views/ValueView.java
package org.cloudfoundry.credhub.views;

import org.cloudfoundry.credhub.credential.StringCredentialValue;
import org.cloudfoundry.credhub.domain.ValueCredentialVersion;

@SuppressWarnings("unused")
public class ValueView extends CredentialView {

  public ValueView() {
    super();
  }

  ValueView(final ValueCredentialVersion valueCredential) {
    super(
      valueCredential.getVersionCreatedAt(),
      valueCredential.getUuid(),
      valueCredential.getName(),
      valueCredential.getCredentialType(),
      new StringCredentialValue((String) valueCredential.getValue())
    );
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/ValueView.kt
package org.cloudfoundry.credhub.views

import org.cloudfoundry.credhub.credential.StringCredentialValue
import org.cloudfoundry.credhub.domain.ValueCredentialVersion

class ValueView : CredentialView {
    constructor() : super() {}
    internal constructor(valueCredential: ValueCredentialVersion) : super(
        valueCredential.versionCreatedAt,
        valueCredential.uuid,
        valueCredential.name,
        valueCredential.getCredentialType(),
        StringCredentialValue((valueCredential.getValue() as String?)!!)
    ) {
    }
}
